src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/MapQueryParameterValueResolver.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class MapQueryParameterValueResolver implements ValueResolverInterface
+{
+    public function resolve(Request $request, ArgumentMetadata $argument): array
+    {
+        $attribute = $argument->getAttributesOfType(MapQueryParameter::class)[0] ?? null;
+        if (null === $attribute) {
+            return [];
+        }
+
+        $name = $attribute->name ?? $argument->getName();
+        if (false === $request->query->has($name)) {
+            if ($argument->isNullable() || $argument->hasDefaultValue()) {
+                return [];
+            }
+
+            throw new NotFoundHttpException(sprintf('Missing query parameter "%s".', $name));
+        }
+
+        if ($argument->isVariadic()) {
+            return $request->query->all($name);
```

Comment:
this one also should validate the type, combining it with FILTER_REQUIRE_ARRAY when doing so

Reply:
Done.